### PR TITLE
_concat_rangeindex_... now returns an empty RangeIndex for empty ranges

### PR DESCRIPTION
**Pull request by SmokinCaterpillar**
This is a follow up to PR #18191.

Overall the code is more readable by adding the constructors at the beginning and
explicitly calling `RangeIndex` constructor.

One range index test was changed to accommodate for the new behavior of the index concatenation function.

- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] 1 test changed

### CODE REVIEW

File: pandas/core/dtypes/concat.py [hunk 1]
```diff
@@ -604,12 +603,8 @@ def _concat_rangeindex_same_dtype(indexes):
...
     else:
         # Here all "indexes" had 0 length, i.e. were empty.
-        # Simply take start, stop, and step from the last empty index.
-        obj = indexes[-1]
-        start = obj._start
-        step = obj._step
-        stop = obj._stop
-
-    return indexes[0].__class__(start, stop, step)
+        # In this case return an empty integer index.
```
Comment (jreback) [inline pandas/core/dtypes/concat.py]: I think that this should be ``RangeIndex(0, 1)`` to be consistent with our default indices, e.g. ``Series([]).index``

Comment (SmokinCaterpillar) [reply pandas/core/dtypes/concat.py]: `RangeIndex(0, 1)` or rather `RangeIndex(0, 0)`?

Comment (jreback) [reply pandas/core/dtypes/concat.py]: sorry yes

Comment (SmokinCaterpillar) [reply pandas/core/dtypes/concat.py]: it is changed to empty range

File: pandas/core/dtypes/concat.py [hunk 2]
```diff
@@ -604,12 +603,8 @@ def _concat_rangeindex_same_dtype(indexes):
         # Get the stop value from "next" or alternatively
         # from the last non-empty index
         stop = non_empty_indexes[-1]._stop if next is None else next
+        return RangeIndex(start, stop, step)
     else:
```
Comment (jreback) [inline pandas/core/dtypes/concat.py]: maybe remove the else here (and just return the RangeIndex)

**Review by jreback — APPROVED** [review]
does this have any user visible effect?

File: pandas/core/dtypes/concat.py (hunk 2 above)
Comment (SmokinCaterpillar) [reply pandas/core/dtypes/concat.py]: ok, I personally like the else in such situations, it's more explicit to me, I guess it's a matter of taste. I'll remove it :-)

**Comment by jreback** [comment]
thanks!

**Pull request merged**